src/OrchardCore.Themes/TheAdmin/Views/Layout.cshtml

```diff
@@ -1,11 +1,13 @@
-@inject DarkModeService DarkModeService;
-@inject IOptions<AdminOptions> AdminOptions;
+@using OrchardCore.Environment.Shell
+@inject DarkModeService DarkModeService
+@inject IOptions<AdminOptions> AdminOptions
 @{
     var adminSettings = Site.As<AdminSettings>();
     var darkMode = await DarkModeService.IsDarkModeAsync();
+    var tenant = Context.Features.Get<ShellContextFeature>().ShellContext.Settings.Name;
```

Comment:
The `ShellContextFeature` is more intended to be used in special cases, e.g. when buiding the tenant pipeline, to retrieve some infos from the ambient request when no more in a given shell scope.

Here better to just resolve / inject `ShellSettings`, so from the actual shell container

Reply:
Not in the view. In the service please 😄

Reply:
Yes you're right, looks better to do it in the service itself

    public string CurrentTenant => _shellSettings.Name;